docs: :memo: major overhaul of iteration guide to reorganize it (#158)
seedcase-project/team: how-we-work/iteration.qmd

-   Will write the title of the iteration as the goal for it, while
    keeping the goal focused, small, acheivable, and considerate of the
    time we have available.
-   Will start adding some tasks into the board.
-   If possible, plan iterations so that they either end on a major
    holiday or are right in the middle of it. Don't plan the start of
    iterations a few days before major holidays start.
-   Will plan and decide on how long the future iterations should be
    (can range from 2-6 weeks, but prefer shorter iterations over longer
    ones). So far, we find that 3 week iterations work the best.

## Start of iteration

::: callout-important
We track our iterations' progress and tasks on GitHub with [this project
board](https://github.com/orgs/seedcase-project/projects/18/).
:::

At the start of the iteration is the first planning meeting that is
scheduled well in advance, for between 1-2 hours and (preferably) right
after the previous iteration's debrief and retrospective. Preferably
this meeting is in person. The aim of the meeting to discuss and plan
the specifics of the iteration.

::: callout-note
A basic agenda for the meeting might be:

> -   Review and agree on the iteration aim and end date

Reviewer: ```suggestion
-   Assign timekeeper.
-   Review and agree on the iteration aim and end date
```

Reviewer: With `.` to fit other lists in this doc

> -   Review and agree on the iteration aim and end date.

Author: ```suggestion
-   Assign a timekeeper.
-   Review and agree on the iteration aim and end date.
```

> -   Review and agree on the iteration aim and end date
> -   Review the list of tasks already listed
> -   Brainstorm and add any other issues as needed
> -   Assign a team member to each task
> -   Schedule the next iteration's end and start meetings
> :::

Reviewer: ```suggestion
-   Review and agree on the iteration aim and end date.
-   Review the list of tasks already listed.
-   Brainstorm and add any other issues as needed.
-   Assign a team member to each task.
-   Schedule the next iteration's end and start meetings.
:::
```
Add `.` to fit other lists in this doc

    relevant for a given task or issue.
-   Work on our assigned tasks and coordinate with others if need be
    (`@` mention the people involved in an issue, don't assign).
-   Add more issues if required. If the issue is relevant to the current
    iteration, then add it to the project board so that we can determine
    who will self-assign it during one of our meetings. If the issue
    isn't relevant, don't add it to the project board but to the
    relevant repository instead, and we will save it for future
    iterations to work on.
-   Plan for "knowledge sharing" / "code review" sessions as necessary,
    for instance to go over a new feature or tool used in a Pull
    Request. For the author of the pull request, they will need to do a
    bit of preparation before hand so that the session runs smoothly.

## End of iteration

::: callout-important
We track our iterations' progress and tasks on GitHub with [this project
board](https://github.com/orgs/seedcase-project/projects/18/).
:::

The iteration's end is a last meeting for a debrief and retrospective to
discuss how things went. These are between 1-2 hours and are preferably

> in person. It will have been scheduled during the planning meeting.

Reviewer: ```suggestion
in person. The time and place of this meeting will have been scheduled during the planning meeting.
```

::: callout-note
A basic agenda is:

-   Assign timekeeper.
-   Demo (internally or externally) the progress of the iteration.
-   Evaluate the iteration aim.
-   Go over what we accomplished and how we felt we did (including
    struggles).
-   Do a retrospective.
:::

During the debrief and retrospective meeting, we will:

> -   Assign a timekeeper to keep us on track and not spending too long on
>     any one topic.

Reviewer: ```suggestion
-   Assign a timekeeper to keep us on track and so we avoid spending too long on
    any one topic.
```

> -   Show off and (optionally) demo the output all together as a team.

Reviewer: ```suggestion
-   Show off and (optionally) demo the output of the iteration all together as a team.
```

    We'll assign someone to be responsible for showing or trying out the
    demo before the meeting.
-   (Optionally done occasionally, but strongly suggested) Include
    someone external (a potential user) to try out the product/demo and
    give some feedback.
-   Evaluate the aim and review whether we've achieved the aim.
-   Briefly go over what we each did during the iteration and how we
    felt we did. Include any barriers, struggles, or challenges that
    came up during the iteration here.
-   Do a "retrospective", where we individually write down how things
    went ("keep doing" category) and what could be improved or removed
    ("stop doing/improve on" category) in the process. The retrospective
    should ideally be done in a collaborative "notetaking/post-it notes"

>     system ([Figma](https://www.figma.com/) is really nice for this).

Reviewer: ```suggestion
    system ([Figma](https://www.figma.com/) is really nice for this). These notes will be collected into topics and we will add action items to each of the topics. The team lead will add these action items to issues or posts on how we work.
```

    (can range from 2-6 weeks, but prefer shorter iterations over longer
    ones). So far, we find that 3 week iterations work the best.

## Start of iteration

::: callout-important
We track our iterations' progress and tasks on GitHub with [this project
board](https://github.com/orgs/seedcase-project/projects/18/).
:::

An iteration starts with a first planning meeting that is scheduled well
in advance, for between 1-2 hours and (preferably) right after the
previous iteration's debrief and retrospective. Preferably this meeting
is in person. The aim of the meeting to discuss and plan the specifics
of the iteration.

::: callout-note
A basic agenda for the meeting might be:

-   Assign a timekeeper.
-   Review and agree on the iteration aim and end date.
-   Review the list of tasks already listed.
-   Brainstorm and add any other issues as needed.
-   Assign a team member to each task.
-   Schedule the next iteration's end and start meetings.
:::

::: panel-tabset
### Before meeting

In general, only the team lead is required to prepare for the planning
meeting. Before the planning meeting, the team lead will move any

> existing tasks (GitHub issues) into the iteration that is relevant to

Author: ```suggestion
existing tasks (GitHub Issues) into the iteration that is relevant to
```

@signekb small thing, when issues are used with the word GitHub, they are capitalized, in line with GitHub's own documentation ("GitHub Issues" vs "create a new issue")

Reviewer: Ah, you're right!
